mergify_engine/rules/filter.py

```diff
@@ -287,3 +304,97 @@ def BinaryFilter(
             "and": all,
         },
     )
+
+
+DT_INFINITY = datetime.datetime.max.replace(tzinfo=datetime.timezone.utc)
+
+
+def _same_datetime(dt: datetime.datetime) -> datetime.datetime:
+    return dt
+
+
+def _minimal_datetime(dts: typing.Iterable[object]) -> datetime.datetime:
+    _dts = list(typing.cast(typing.List[datetime.datetime], Filter._to_list(dts)))
+    if len(_dts) == 0:
+        return DT_INFINITY
+    else:
+        return min(_dts)
+
+
+def _as_datetime(value: typing.Any) -> datetime.datetime:
+    if isinstance(value, datetime.datetime):
+        return value
+    elif isinstance(value, datetime.time):
+        return utils.utcnow().replace(
+            hour=value.hour,
+            minute=value.minute,
+            second=value.second,
+            microsecond=value.microsecond,
+        )
+    else:
+        return DT_INFINITY
+
+
+def _dt_max(value: typing.Any, ref: typing.Any) -> datetime.datetime:
+    return DT_INFINITY
+
+
+def _dt_op(
+    op: typing.Callable[[typing.Any, typing.Any], bool],
+) -> typing.Callable[[typing.Any, typing.Any], datetime.datetime]:
+    def _operator(value: typing.Any, ref: typing.Any) -> datetime.datetime:
+        dt_value = _as_datetime(value)
+        dt_ref = _as_datetime(ref)
+        handle_equality = op in (operator.eq, operator.ne, operator.le, operator.ge)
+        if handle_equality and dt_value == dt_ref:
+            # It will change very soon!
+            try:
+                return dt_ref + datetime.timedelta(minutes=5)
+            except OverflowError:
+                return dt_ref
+        elif dt_value < dt_ref:
+            return dt_ref
+        elif isinstance(ref, datetime.time):
+            # Condition will change, next day at 00:00:00
+            try:
+                dt_ref = dt_ref + datetime.timedelta(days=1)
+            except OverflowError:
+                return DT_INFINITY
+            if op in (operator.eq, operator.ne):
+                return dt_ref
+            else:
+                return dt_ref.replace(hour=0, minute=0, second=0, microsecond=0)
+        else:
+            return DT_INFINITY
+
+    return _operator
+
+
+def NearDatetimeFilter(
+    tree: typing.Union[TreeT, CompiledTreeT[GetAttrObject, datetime.datetime]],
+) -> "Filter[datetime.datetime]":
+    """
+    The principe:
+    * the attribute can't be mapped to a datetime -> datetime.datetime.max
+    * the time/datetime attribute can't change in the future -> datetime.datetime.max
+    * the time/datetime attribute can change in the future -> return when
+    * we have a list of time/datetime, we pick the more recent
+    """
+    return Filter[datetime.datetime](
+        tree,
+        {"-": _same_datetime},
```

Comment:
That does not sound right. I was trying to look for a test to understand how that worked and I didn't find one. 👎  😁 

Also can't you use `_identity`? If it's a typing issue it should work with a `TypeVar`.

Reply:
I added a test